sdk/digitaltwins/azure-digitaltwins-core/src/test/java/com/azure/digitaltwins/core/ComponentsTests.java

```diff
@@ -0,0 +1,91 @@
+        DigitalTwinsClient client = getClient(httpClient, serviceVersion);
+
+        String wifiComponentName = "wifiAccessPoint";
+
+        String roomWithWifiTwinId = UniqueIdHelper.getUniqueDigitalTwinId(TestAssetDefaults.ROOM_WITH_WIFI_TWIN_ID_PREFIX, client, randomIntegerStringGenerator);
+        String roomWithWifiModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.ROOM_WITH_WIFI_MODEL_ID_PREFIX, client, randomIntegerStringGenerator);
+        String wifiModelId = UniqueIdHelper.getUniqueModelId(TestAssetDefaults.WIFI_MODEL_ID_PREFIX, client, randomIntegerStringGenerator);
+
+        String modelWifi = TestAssetsHelper.getWifiModelPayload(wifiModelId);
+        String modelRoomWithWifi = TestAssetsHelper.getRoomWithWifiModelPayload(roomWithWifiModelId, wifiModelId, wifiComponentName);
+        String roomWithWifiTwin = TestAssetsHelper.getRoomWithWifiTwinPayload(roomWithWifiModelId, wifiComponentName);
+
+        List<String> modelsList = new ArrayList<>(Arrays.asList(modelWifi, modelRoomWithWifi));
+
+        try {
+            // Create models and components to test the lifecycle.
+            List<ModelData> createdList = client.createModels(modelsList);
+            logger.info("Created {} models successfully", createdList.size());
+
+            BasicDigitalTwin createdTwin = client.createDigitalTwin(roomWithWifiTwinId, roomWithWifiTwin,BasicDigitalTwin.class);
+
+            logger.info("Created {} twin successfully", createdTwin.getId());
+            assertEquals(createdTwin.getId(), roomWithWifiTwinId);
+
+            // Get the component
+            Response<String> getComponentResponse = client.getComponentWithResponse(roomWithWifiTwinId, wifiComponentName, Context.NONE);
+            assertEquals(getComponentResponse.getStatusCode(), HttpURLConnection.HTTP_OK);
+
+            // Update component
+            DigitalTwinsResponse<Void> upDateComponentResponse = client.updateComponentWithResponse(
+                roomWithWifiTwinId,
+                wifiComponentName,
+                TestAssetsHelper.getWifiComponentUpdatePayload(),
+                new UpdateComponentRequestOptions(),
+                Context.NONE);
+
+            assertEquals(upDateComponentResponse.getStatusCode(), HttpURLConnection.HTTP_NO_CONTENT);
+        }
+        // clean up
+        finally {
+            try
+            {
+                if (roomWithWifiTwinId != null)
+                {
+                    client.deleteDigitalTwin(roomWithWifiTwinId);
+                }
+                if (roomWithWifiModelId != null)
+                {
+                    client.deleteModel(roomWithWifiModelId);
+                }
+                if (wifiModelId != null)
+                {
+                    client.deleteModel(wifiModelId);
+                }
+            }
+            catch (Exception ex)
+            {
+                fail("Test clean up failed: " + ex.getMessage());
```

Comment:
To preserve the full stacktrace, how about:

throw new AssertionFailedError("Test clean up failed", ex)